openedx/core/djangoapps/course_groups/models.py

```diff
@@ -36,9 +36,26 @@ class Meta:
     GROUP_TYPE_CHOICES = ((COHORT, 'Cohort'),)
     group_type = models.CharField(max_length=20, choices=GROUP_TYPE_CHOICES)
 
+    @classmethod
+    def create(cls, name, course_id, group_type=COHORT):
+        """
+        Create a new course user group.
+
+        Args:
+            name: Name of group
+            course_id: course id
+            group_type: group type
+        """
+        return cls.objects.get_or_create(
+            course_id=course_id,
+            group_type=group_type,
+            name=name
+        )
```

Comment:
the idea i had in mind with my earlier suggestion was to put get_or_create for both CourseUserGroup and CourseCohort inside this method - does that seem like a good fit to you?